sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Canonicalize.scala

```diff
@@ -42,7 +42,7 @@ object Canonicalize {
   /** Remove names and nullability from types, and names from `GetStructField`. */
   private[expressions] def ignoreNamesTypes(e: Expression): Expression = e match {
     case a: AttributeReference =>
-      AttributeReference("none", a.dataType.asNullable)(exprId = a.exprId)
+      AttributeReference("none", a.dataType)(exprId = a.exprId)
```

Comment:
Ur, I realized that we should fix line 42 too. Could you fix it master branch first as a follow-up of SPARK-38030, @shardulm94 ?
```
- Remove names and nullability from types
+ Remove names from types
```

Reply:
Good catch! The canonicalization code has change substantiantially between 3.2 and master due to https://github.com/apache/spark/pull/34883. This comment no longer exists in the master branch so it should not be an issue. I will fix it in 3.2 and 3.1 PRs.

Reply:
Got it. Thank you for checking, @shardulm94 .